lib/index.ml

```diff
@@ -71,18 +104,39 @@ let get_job_ids t ~owner ~name ~hash =
   | Sqlite3.Data.[ TEXT variant; TEXT id ] -> variant, Some id
   | row -> Fmt.failwith "get_job_ids: invalid row %a" Db.dump_row row
 
-let record ~repo ~hash jobs =
+let get_status ~owner ~name ~hash =
+  let t = Lazy.force db in
+  let open Sqlite3.Data in
+  Db.query t.get_status [ TEXT owner; TEXT name; TEXT hash ]
+  |> function
+    | [ [ INT 0L ] ] -> `Pending
+    | [ [ INT 1L ] ] -> `Failed
+    | [ [ INT 2L ] ] -> `Passed
+    | [ [] ] -> `Not_started
+    | rows -> Fmt.failwith "get_status: invalid rows %a" Fmt.(list ~sep:cut Db.dump_row) rows
+
+let record ~repo ~hash ~status jobs =
   let { Current_github.Repo_id.owner; name } = repo in
   let t = Lazy.force db in
+  let () =
+    let open Sqlite3.Data in
+    let status =
+      match status with
+      | `Pending -> INT 0L
+      | `Failed -> INT 1L
+      | `Passed -> INT 2L
+    in
+    Db.exec t.record_status [ TEXT owner; TEXT name; TEXT hash; status ];
```

Comment:
I'll bear this in mind when testing/deploying. We could perhaps just keep this state in memory? It's going to be regenerated on each deploy anyway.

Having said that, I agree that this problem (and likely many others) could just be fixed with a cleverer pipeline evaluation strategy.

Reply:
Keeping it in memory does sound like a better idea. The commits have to be persisted because you can click on historical commits in GitHub and it needs to take you to the archived logs, but there's no reason to track out-of-date badges.

Reply:
Done in https://github.com/ocurrent/ocaml-ci/pull/100/commits/34b0d05b3ca6c017243726c178628d233266d355.